x/observer/keeper/utils.go

```diff
@@ -147,3 +147,33 @@ func (k Keeper) CheckObserverSelfDelegation(ctx sdk.Context, accAddress string)
 	}
 	return nil
 }
+
+// VoteOnBallot finds a ballot or creates a new one if not found,
```

Comment:
This is not directly related to this issue as it was already the case but in general we should avoid utils.go file name, maybe we can move this one and other vote related method to a `voting.go` file?